src/services/action/utils/export.ts

```diff
@@ -130,24 +141,58 @@ export const exportActionsInArchive = async (args: {
     const chatPath = path.join(fileFolderPath, buildActionFileName('chat', archiveDate, format));
     writeFileForFormat(chatPath, format, baseAnalytics.chatMessages);
 
-    // create file for the apps
-    const appsPath = path.join(fileFolderPath, buildActionFileName('apps', archiveDate, format));
-    writeFileForFormat(
-      appsPath,
-      format,
-      // get actions from apps data
-      Object.entries(baseAnalytics.apps)
-        .map(([appID, { actions, data, settings }]) =>
-          actions.map((action) => ({
-            ...action,
-            appID,
-            data: JSON.stringify(data),
-            settings: JSON.stringify(settings),
-          })),
-        )
-        .flat(),
+    // merge together actions, data and settings from all app_items
+    const { appActions, appData, appSettings } = Object.entries(baseAnalytics.apps).reduce<{
+      appActions: AppAction[];
+      appData: AppData[];
+      appSettings: AppSetting[];
+    }>(
+      (acc, [_appID, { actions, data, settings }]) => {
+        acc.appActions.push(...actions);
+        acc.appData.push(...data);
+        acc.appSettings.push(...settings);
+        return acc;
+      },
+      { appActions: [], appData: [], appSettings: [] },
     );
 
+    switch (format) {
+      // For JSON format only output a single file
+      case ExportActionsFormatting.JSON: {
+        // create files for the apps
+        const appsPath = path.join(
+          fileFolderPath,
+          buildActionFileName('apps', archiveDate, format),
+        );
+        writeFileForFormat(appsPath, format, { appActions, appData, appSettings });
+        break;
+      }
+      // For CSV format there will be one file for actions, one fofr data and one for settings
+      // with all the apps together.
+      case ExportActionsFormatting.CSV: {
+        // create files for the apps
+        const appActionsPath = path.join(
+          fileFolderPath,
+          buildActionFileName('app_actions', archiveDate, format),
+        );
+        writeFileForFormat(appActionsPath, format, appActions);
+
+        const appDataPath = path.join(
+          fileFolderPath,
+          buildActionFileName('app_data', archiveDate, format),
+        );
+        writeFileForFormat(appDataPath, format, appData);
+
+        const appSettingsPath = path.join(
+          fileFolderPath,
+          buildActionFileName('app_settings', archiveDate, format),
+        );
+        writeFileForFormat(appSettingsPath, format, appSettings);
```

Comment:
Should we refactor in another function to then call it for the three CSV exports, as well as for the JSON export?

Reply:
I don't know, is the function too lengthy like this ? Not sure refactoring to a different function will make it more readable.

Reply:
I'm okay with what your did too ^^